File: src/main/java/com/python/array_index_and_element.py
class Solution:
    def array_index_and_element(self, arr):
        if len(arr) == 1:
            return 0 if arr[0] == 0 else -1

        left = 0
        right = len(arr) - 1
        result = -1

        while left <= right:
            mid = left + (right - left) // 2
            val = arr[mid]

            if val == mid:
                result = mid
                right = mid - 1
            elif val < mid:
                left = mid + 1
            else:
                right = mid - 1

        return result

File: src/main/java/com/ArrayIndexAndElement.java
public class ArrayIndexAndElement {
    int indexEqualsValueSearch(int[] arr) {
        if (arr.length == 1) {
            return (arr[0] == 0) ? 0 : -1;

        }
        int left = 0;
        int right = arr.length - 1;
        int result = -1;

        while (left <= right) {
            int mid = left + (right - left) / 2;
            int value = arr[mid];

            if (value == mid) {
                result = mid;
                right = mid - 1;
            } else if (value < mid) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return result;
    }
}
